R/arc-select.R

```diff
@@ -454,10 +450,91 @@ count_results <- function(req, query) {
   resp <- httr2::resp_body_string(
     httr2::req_perform(
       httr2::req_url_query(n_req, f = "json"),
-      error_call = rlang::caller_env()
+      error_call = error_call
     )
   )
 
   RcppSimdJson::fparse(resp)[["count"]]
 }
 
+#' Set page size and check page size validity
+#'
+#' @noRd
+set_page_size <- function(
+    page_size = NULL,
+    max_records = 25000L,
+    error_call = rlang::caller_env()
+    ) {
+  # if its null, just use max records (default)
+  if (is.null(page_size)) {
+    return(max_records)
+  }
+
```

Comment:
We need to ensure that the type is numeric. Because this might error or return `NA` which is not checked later
```suggestion
if (!is.numeric(page_size)) {
    cli::cli_abort("{.arg page_size} must be a numeric scalar not {.obj_type_friendly {class(page_size)}}", call = error_call)
}
```